### PR TITLE
Use event bus instead of core client

### PR DESCRIPTION
**Pull request by sophiemachin**
## Motivation

Move towards fully event-based architecture by removing any direct calls to other stacks, and using the EventBus instead

## Source code

`appointments.py` and `views.py` have been updated to put an event on the EventBus instead of using the core api client to call api endpoints. 

## Tests

Tests no longer check the outcome of lambdas in other stacks (ie dont wait and check the dynamo db table)

### CODE REVIEW

**Comment by this-andy** [comment]
https://this.tpondemand.com/entity/4818

File: tests/testing_utilities.py [hunk 1]
```diff
@@ -37,13 +37,11 @@
...
+
     @classmethod
     def clear_appointments_table(cls):
-        try:
-            cls.ddb_client.delete_all(table_name=app.APPOINTMENTS_TABLE)
-        except AttributeError:
-            cls.ddb_client = Dynamodb(stack_name=STACK_NAME)
-            cls.ddb_client.delete_all(table_name=app.APPOINTMENTS_TABLE)
+        ddb_client = Dynamodb(stack_name=STACK_NAME)
+        ddb_client.delete_all(table_name=app.APPOINTMENTS_TABLE)
```
Comment (alistairclark) [inline tests/testing_utilities.py:44]: 👍

**Review by alistairclark — APPROVED** [review]
All looking food. There's some logging in the thiscovery-core transactional email method that we aren't brining across here. I'm guessing that's fine, as we probably don't want to dump too much info about emails in logs, but just thought I'd flag in case you missed.

File: src/reminders.py [hunk 2]
```diff
@@ -89,8 +89,8 @@ def send_reminders(self, now=None):
             )
 
             try:
-                reminder_result = notifier.send_reminder(now=now).get("statusCode")
-            except:
+                reminder_result = notifier.send_reminder(now=now)
+            except Exception as e:
```
[the remark this answers is not in the document]
Comment (alistairclark) [reply src/reminders.py:93]: Worth specifying the exceptions we're catching?

**Review by alistairclark — APPROVED** [review]
Looks good. In general I would always try to specify exceptions we're catching, even if there are a few, but approving anyway.

**Review by this-andy — APPROVED** [review]

File: src/reminders.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (sophiemachin) [reply src/reminders.py:93]: Yeah I agree, but also I don't actually know what exceptions we're intending to catch. Will chat at our standup.

[the remark this answers is not in the document]
Comment (sophiemachin) [reply src/reminders.py:93]: I think this was originally checking for errors from the core_client 

"Raises an AssertionError if the returned status code is unexpected."

**Pull request merged**